Modules/Core/Common/src/itkExtractImageFilter.cxx

```diff
@@ -0,0 +1,46 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include "itkExtractImageFilter.h"
+
+namespace itk
+{
+/** Define how to print enumeration */
+std::ostream &
+operator<<( std::ostream & out, const ExtractImageFilterCollapseStrategy value )
+{
+  const char * s = nullptr;
+  switch ( value )
+  {
+    case ExtractImageFilterCollapseStrategy::DIRECTIONCOLLAPSETOUNKOWN:
```

Comment:
I'd use a macro for each case where only "DIRECTIONCOLLAPSETOUNKOWN" is needed. This would eliminate any discrepancy between the enum name and the string.

Reply:
@blowekamp I don't understand what you are proposing.  Can you give some pseudocode?  Could that be a second PR later?